torch/utils/cpp_extension.py

```diff
@@ -1484,7 +1486,7 @@ def _jit_compile(name,
         baton = FileBaton(os.path.join(build_directory, 'lock'))
         if baton.try_acquire():
             try:
-                with GeneratedFileCleaner(keep_intermediates=keep_intermediates) as clean_ctx:
+                with hipify_python.GeneratedFileCleaner(keep_intermediates=keep_intermediates) as clean_ctx:
```

Comment:
@lcskrishna We will need to figure out a way to:

1. either add a `catch` block for the error thrown by this line on CUDA builds, since `hipify_python` is set to `None` here: https://github.com/pytorch/pytorch/pull/101391/files#diff-258d5272916c8cf5ac89aad77f7c114b585848826f6bbaeee91fe62a4391ee5eR22... However, this may not be required if we end up installing hipify_torch for CUDA builds as well, due to the other failure wrt test_testing
2. conditionalize the use of this context manager to use it just for ROCm builds